File: app/models/mbom.py
from sqlalchemy import Column, String, BigInteger, Numeric, DateTime
from sqlalchemy import UniqueConstraint, Index
from sqlalchemy.orm import relationship
from app.core.database import Base


class Mbom(Base):
    """レシピマスタ/BOM（MBOM）- 全33カラム定義"""

    __tablename__ = "mbom"

    # プライマリキー
    prkey = Column(BigInteger, primary_key=True, comment="プライマリキー")

    # 親品目情報
    pfctcd = Column(String(50), nullable=False, default="", comment="親工場コード")
    pdeptcd = Column(String(50), nullable=False, default="", comment="親部門コード")
    pitemgr = Column(String(1), nullable=False, default="", comment="親品目グループ")
    pitemcd = Column(String(50), nullable=False, default="", comment="親品目コード")
    proutno = Column(Numeric, nullable=False, default=0, comment="")
    pcd1 = Column(Numeric, nullable=False, default=0, comment="")
    pcd2 = Column(Numeric, nullable=False, default=0, comment="")

    # 子品目情報
    cfctcd = Column(String(50), nullable=False, default="", comment="子工場コード")
    cdeptcd = Column(String(50), nullable=False, default="", comment="子部門コード")
    citemgr = Column(String(1), nullable=False, default="", comment="子品目グループ")
    citemcd = Column(String(50), nullable=False, default="", comment="子品目コード")

    # 数量・タイプ
    amu = Column(Numeric, default=0, comment="")
    otp = Column(Numeric, default=0, comment="")
    partyp = Column(String(1), default="", comment="")
    par = Column(Numeric, default=0, comment="")
    prvtyp = Column(String(1), default="", comment="")

    # 発行条件
    issjor = Column(String(150), default="", comment="")
    memo = Column(String(150), default="", comment="メモ")
    mbompic = Column(String(50), default="", comment="")

    # 期間
    stadt = Column(String(8), nullable=False, default="", comment="開始日")
    enddt = Column(String(8), nullable=False, default="", comment="終了日")

    # システム情報
    uuser = Column(String(20), default="", comment="更新ユーザー")
    udate = Column(DateTime, comment="更新日時")

    # 設定フラグ
    unset = Column(String(1), default="", comment="")
    bbdtset = Column(String(1), default="0", comment="")
    planruntgtflg = Column(String(1), nullable=False, default="0", comment="")

    # 重量・グループ
    weighqun = Column(Numeric, default=0, comment="")
    throwngrpno = Column(Numeric, default=0, comment="")
    ratio = Column(Numeric, default=0, comment="")
    weightyp = Column(String(1), default="", comment="")

    # 追加情報
    addfst = Column(Numeric, default=0, comment="")
    mbomlt = Column(Numeric, default=0, comment="")

    # 制約定義
    __table_args__ = (
        UniqueConstraint(
            "pfctcd",
            "pdeptcd",
            "pitemgr",
            "pitemcd",
            "proutno",
            "pcd1",
            "pcd2",
            "cfctcd",
            "cdeptcd",
            "citemgr",
            "citemcd",
            "stadt",
            "enddt",
            name="mbom_pfctcd_pdeptcd_pitemgr_pitemcd_proutno_pcd1_pcd2_cfctc_key",
        ),
        Index("mbom_mbompic", "mbompic"),
        Index(
            "mbom_pfctcd_pdeptcd_pitemgr_pitemcd_stadt_enddt",
            "pfctcd",
            "pdeptcd",
            "pitemgr",
            "pitemcd",
            "stadt",
            "enddt",
        ),
    )

    # ========================================
    # リレーション定義
    # ========================================

    # 子品目マスタ（Item）
    child_item = relationship(
        "Item",
        primaryjoin=(
            "and_("
            "foreign(Mbom.cfctcd) == Item.fctcd, "
            "foreign(Mbom.cdeptcd) == Item.deptcd, "
            "foreign(Mbom.citemgr) == Item.itemgr, "
            "foreign(Mbom.citemcd) == Item.itemcd"
            ")"
        ),
        viewonly=True,
        lazy="select",
    )

    def __repr__(self):
        return f"<Mbom(prkey={self.prkey}, pitemcd={self.pitemcd}, citemcd={self.citemcd})>"

File: src/BaggingInstructions.Api/Entities/Mbom.cs
using System.ComponentModel.DataAnnotations.Schema;
using BaggingInstructions.Api.Entities.Legacy;

namespace BaggingInstructions.Api.Entities.Legacy;

[Table("mbom")]
public class Mbom
{
    [Column("prkey")]
    public long Prkey { get; set; }

    [Column("pfctcd")]
    public string Pfctcd { get; set; } = "";

    [Column("pdeptcd")]
    public string Pdeptcd { get; set; } = "";

    [Column("pitemgr")]
    public string Pitemgr { get; set; } = "";

    [Column("pitemcd")]
    public string Pitemcd { get; set; } = "";

    [Column("proutno")]
    public decimal Proutno { get; set; }

    [Column("pcd1")]
    public decimal Pcd1 { get; set; }

    [Column("pcd2")]
    public decimal Pcd2 { get; set; }

    [Column("cfctcd")]
    public string Cfctcd { get; set; } = "";

    [Column("cdeptcd")]
    public string Cdeptcd { get; set; } = "";

    [Column("citemgr")]
    public string Citemgr { get; set; } = "";

    [Column("citemcd")]
    public string Citemcd { get; set; } = "";

    [Column("amu")]
    public decimal Amu { get; set; }

    [Column("otp")]
    public decimal Otp { get; set; }

    [Column("partyp")]
    public string? Partyp { get; set; } = "";

    [Column("par")]
    public decimal? Par { get; set; }

    [Column("prvtyp")]
    public string? Prvtyp { get; set; } = "";

    [Column("issjor")]
    public string? Issjor { get; set; } = "";

    [Column("memo")]
    public string? Memo { get; set; } = "";

    [Column("mbompic")]
    public string? Mbompic { get; set; } = "";

    [Column("stadt")]
    public string Stadt { get; set; } = "";

    [Column("enddt")]
    public string Enddt { get; set; } = "";

    [Column("uuser")]
    public string? Uuser { get; set; } = "";

    [Column("udate")]
    public DateTime? Udate { get; set; }

    [Column("unset")]
    public string? Unset { get; set; } = "";

    [Column("bbdtset")]
    public string? Bbdtset { get; set; } = "0";

    [Column("planruntgtflg")]
    public string Planruntgtflg { get; set; } = "0";

    [Column("weighqun")]
    public decimal Weighqun { get; set; }

    [Column("throwngrpno")]
    public decimal Throwngrpno { get; set; }

    [Column("ratio")]
    public decimal Ratio { get; set; }

    [Column("weightyp")]
    public string? Weightyp { get; set; } = "";

    [Column("addfst")]
    public decimal Addfst { get; set; }

    [Column("mbomlt")]
    public decimal Mbomlt { get; set; }

    // Navigation: 子品目マスタ
    public virtual ItemLegacy? ChildItem { get; set; }
}
